### PR TITLE
solution for make it one

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

don't worry, i'm still active as a CA myself

### CODE REVIEW

File: solutions/gold/cf-1043F.mdx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+constexpr int MOD = 1e9 + 7;
+constexpr int ANS_UB = 7;
+
+long long pow(long long base, long long exp) {
```
Comment (envyaims) [inline solutions/gold/cf-1043F.mdx:141]: maybe dont name it pow since thats a built in c++ function name

File: solutions/gold/cf-1043F.mdx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+			long long val = comb(div_by_num[j], i);
+			for (int j_mul = 2 * j; j_mul <= max; j_mul += j) {
+				val = (val - gcd_ways[j_mul] + MOD) % MOD;
+			}
+			gcd_ways[j] = val;
+		}
+
+		if (gcd_ways[1] > 0) {
+			cout << i << endl;
+			goto end;
```
Comment (envyaims) [inline solutions/gold/cf-1043F.mdx:193]: cant you just return?

**Review by envyaims — CHANGES_REQUESTED** [review]

File: solutions/gold/cf-1043F.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/gold/cf-1043F.mdx:141]: precisely why i don't do `using namespace std`

Comment (envyaims) [reply solutions/gold/cf-1043F.mdx:141]: still ambiguous

**Review by envyaims — APPROVED** [review]

File: solutions/gold/cf-1043F.mdx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+n = int(input())
+
+nums = {int(i) for i in input().split()}
+max_ = max(nums)
+
+div_by_num = [0 for _ in range(max_ + 1)]
+for d in range(1, max_ + 1):
+	for mul in range(d, max_ + 1, d):
+		div_by_num[d] += mul in nums
```
Comment (ryanchou-dev) [inline solutions/gold/cf-1043F.mdx:106]:
```suggestion
		div_by_num[d] += (mul in nums)
```
helps readability

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**